### ISSUE TITLE
PCC results numerlcally identical to torchdr for torchdr > 0.2

### ISSUE THREAD

**Issue by e-pet** [jacobgil/pcc#6]
Hi Jacob,

in the following example, I am seeing clearly different results between pcc and directly calling torchdr.umap for torchdr=0.2 (as expected / desired) but numerically identical outputs between the two for torchdr=0.3 or 0.4.

There seem to have been some API changes in torchdr across those versions (e.g. I noticed that `metric_in` is now just called `metric`), so I suspect some aspect of the pcc -> torchdr routing fails for newer versions?

```{python}
import matplotlib.pyplot as plt, numpy as np, pcc, torchdr

x = np.load("odelia_pcc_demo_pca.npy")
xn = x / np.linalg.norm(x, axis=1, keepdims=True)

args = {'n_neighbors': 30, 'n_components': 3, 'random_state': 42, 'device': 'cpu', 'backend': None}

if torchdr.__version__ == "0.2":
    pcumap = pcc.PCUMAP(**args, metric_in="euclidean")
elif torchdr.__version__ == "0.3":
    pcumap = pcc.PCUMAP(**args, metric_in="euclidean")
elif torchdr.__version__ == "0.4":
    pcumap = pcc.PCUMAP(**args, metric="euclidean")

y = pcumap.fit_transform(xn)

fig = plt.figure(figsize=(7, 7)); ax = fig.add_subplot(projection="3d"); ax.scatter(y[:, 0], y[:, 1], y[:, 2], s=4)
fig.savefig(f"odelia_pcc_demo_torchdr_{torchdr.__version__.replace('.', '_')}.jpg", dpi=200, bbox_inches="tight")

if torchdr.__version__ == "0.2":
    umap = torchdr.UMAP(**args, metric_in="euclidean")
elif torchdr.__version__ == "0.3":
    umap = torchdr.UMAP(**args, metric_in="euclidean")
elif torchdr.__version__ == "0.4":
    umap = torchdr.UMAP(**args, metric="euclidean")

z = umap.fit_transform(xn)
d = np.abs(y - z)
# allclose is False for torchdr 0.2 but True for torchdr 0.3 & 0.4
print("allclose", np.allclose(y, z), "max_abs", d.max(), "mean_abs", d.mean())

fig = plt.figure(figsize=(7, 7)); ax = fig.add_subplot(projection="3d"); ax.scatter(z[:, 0], z[:, 1], z[:, 2], s=4)
fig.savefig(f"odelia_umap_demo_torchdr_{torchdr.__version__.replace('.', '_')}.jpg", dpi=200, bbox_inches="tight")
```

package versions (only torchdr version changes between runs):
```
Package           Version
----------------- -----------
colorama          0.4.6
contourpy         1.3.2
cycler            0.12.1
filelock          3.29.0
fonttools         4.62.1
fsspec            2026.3.0
Jinja2            3.1.6
joblib            1.5.3
kiwisolver        1.5.0
MarkupSafe        3.0.3
matplotlib        3.10.8
mpmath            1.3.0
networkx          3.4.2
numpy             2.2.6
packaging         26.1
pccdr             1.0.3
pillow            12.2.0
pip               26.0.1
pyparsing         3.3.2
python-dateutil   2.9.0.post0
scikit-learn      1.4.2
scipy             1.15.3
setuptools        81.0.0
six               1.17.0
sympy             1.14.0
threadpoolctl     3.6.0
torch             2.11.0
torch-tb-profiler 0.4.3
torchdr           0.4
tqdm              4.67.3
typing_extensions 4.15.0
wheel             0.47.0
```

torchdr 0.2, pcc:
<img width="200" height="200" alt="Image" src="https://github.com/user-attachments/assets/517702ff-14ce-41d8-9577-a83d24837626" />

torchdr 0.2, plain torchdr:
<img width="200" height="200" alt="Image" src="https://github.com/user-attachments/assets/e9040898-2d1d-43b0-8e86-9996ff0e289d" />

torchdr 0.3, pcc:
<img width="200" height="200" alt="Image" src="https://github.com/user-attachments/assets/81a950f5-4a15-446f-aeb8-04e721df0208" />

torchdr 0.3, plain torchdr:
<img width="200" height="200" alt="Image" src="https://github.com/user-attachments/assets/5f91b91e-7596-473e-bd71-3b5f14ceb24a" />

torchdr 0.4, pcc:
<img width="200" height="200" alt="Image" src="https://github.com/user-attachments/assets/429b66b5-7624-447c-9bcb-d1922dcbd6c6" />

torchdr 0.4, plain torchdr:
<img width="200" height="200" alt="Image" src="https://github.com/user-attachments/assets/bb6e8919-aff3-40e9-bccd-571fb7de732f" />

Input data if you want to reproduce this particular example: [odelia_pcc_demo_pca.npy.zip](https://github.com/user-attachments/files/27008206/odelia_pcc_demo_pca.npy.zip)
